ttkwidgets/tickscale.py

```diff
@@ -0,0 +1,633 @@
+                raise ValueError("'labelpos' must be 'n', 's', 'e' or 'w'.")
+            else:
+                self._labelpos = labelpos
+                reinit = True
+        if 'resolution' in kw:
+            try:
+                self._resolution = float(kw.pop('resolution'))
+                if self._resolution < 0:
+                    raise ValueError("'resolution' must be non negative.")
+            except ValueError:
+                raise TypeError("'resolution' must be a float.")
+        if self._tickinterval != 0 and self._resolution > self._tickinterval:
+            self._tickinterval = self._resolution
+            reinit = True
+        if 'digits' in kw:
+            digits = kw.pop('digits')
+            if not isinstance(digits, int):
+                raise TypeError("'digits' must be an integer.")
+            elif digits < 0:
+                self._digits = digits
+                self._formatter = '{:g}'
+                reinit = True
+            else:
+                self._digits = digits
+                self._formatter = '{:.' + str(self._digits) + 'f}'
+                interv = self._get_precision(self._tickinterval)
+                resol = self._get_precision(self._resolution)
+                start = kw.get('from', kw.get('from_', self._start))
+                end = kw.get('to', self.scale.cget('to'))
+                from_ = self._get_precision(start)
+                to = self._get_precision(end)
+                d = max(interv, resol, from_, to)
+                if self._digits < d:
+                    self._resolution = float('1e-{}'.format(self._digits))
+                    self._tickinterval = round(self._tickinterval, self._digits)
+                    if self._resolution > self._tickinterval:
+                        self._tickinterval = self._resolution
+                    kw['to'] = round(end, self._digits)
+                    if 'from_' in kw:
+                        del kw['from_']
+                    kw['from'] = round(start, self._digits)
+                reinit = True
+        elif self._digits > 0:
+            start = kw.get('from', kw.get('from_', self._start))
+            end = kw.get('to', self.scale.cget('to'))
+            from_ = self._get_precision(start)
+            to = self._get_precision(end)
+            interv = self._get_precision(self._tickinterval)
+            resol = self._get_precision(self._resolution)
+            digits = max(self._digits, interv, resol, from_, to)
+            if digits != self._digits:
+                self._digits = digits
+                self._formatter = '{:.' + str(self._digits) + 'f}'
+                reinit = True
+        if 'variable' in kw:
+            self._var = kw['variable']
+            if not self._var:
+                self._var = tk.DoubleVar(self, self.get())
+                kw['variable'] = self._var
+            try:
+                self._var.trace_add('write', self._increment)
+            except Exception:
```

Comment:
Could this except clause be made smaller by catching a `TclError`?

Reply:
I have checked, it is an `AttributeError` which will be raised, I will correct it.